templates/types/streaming/fastapi/app/api/routers/upload.py

```diff
@@ -14,22 +15,31 @@
 class FileUploadRequest(BaseModel):
     base64: str
     filename: str
+    filesize: Optional[int] = None
+    filetype: Optional[str] = None
     params: Any = None
 
 
 @r.post("")
-def upload_file(request: FileUploadRequest) -> Dict[str, Any]:
+def upload_file(request: FileUploadRequest) -> DocumentFile:
     """
     To upload a private file from the chat UI.
-    Returns:
-        The metadata of the uploaded file.
     """
     try:
         logger.info(f"Processing file: {request.filename}")
         file_meta = PrivateFileService.process_file(
             request.filename, request.base64, request.params
         )
-        return file_meta.to_upload_response()
+
+        document_file = DocumentFile(
+            id=file_meta.id,
+            metadata=file_meta,
+            # Still return the original fields of the request to display in the chat UI.
+            filename=request.filename,
+            filesize=request.filesize,
+            filetype=request.filetype,
```

Comment:
should be easy to get file size on the server side - mime type can also be determined but might have an advantage to do this one the client side